### DISCUSSION TITLE
Load spline events from Scene into r3f

### DISCUSSION THREAD

**Discussion by guerragiuseppe1507** [splinetool/r3f-spline#19] [Q&A] [answered]
When spline generate r3f code does not import events and states. Is possible to manually import events and states from spline scene into r3f?

Printing nodes i see that events and states are in the object but i'm wondering if there are some function to import them or i have to build from scratch

<img width="683" alt="image" src="https://github.com/splinetool/r3f-spline/assets/12450060/9b220bbd-8e83-424c-96d8-251ee422afa5">

**Comment by superguigui** [comment, marked as answer]
Hi ! Currently spline events are not supported in R3F exports and three.js exports.
There are various reasons for that, mainly because those events rely on Spline runtime and the r3f export is using a simpler loader.
In r3f export it's assumed for the user to manually handle events from react code instead which can be an advantage in some situation where you need more control.
If you want to use spline events in react then you can use the react export which relies on Spline runtime.
